target/linux/realtek/files-5.10/drivers/clk/realtek/clk-rtl83xx.c

```diff
@@ -484,6 +481,26 @@ static const struct clk_ops rtcl_clk_ops = {
 	.recalc_rate = rtcl_recalc_rate,
 };
 
+static void rtcl_unlock_registers(void)
+{
+	u32 enable, reg;
+
+	if (rtcl_ccu->soc != RTCL_SOC838X)
+		return;
+/*
+ * Some RTL838X devices are shipped with register access locked. In this case
+ * we cannot read and/or write to LXB or SW PLL registers. As there is no real
+ * benefit of always unlocking/locking these registers just open up everything.
+ */
+	enable = RTL838X_INT_RW_CTRL_READ_EN | RTL838X_INT_RW_CTRL_WRITE_EN;
+	reg = ioread32((void *)RTL838X_INT_RW_CTRL);
```

Comment:
This crashes my device on boot, since it's missing the switch core address space offset.

Reply:
fixed, thanks for testing